pandas/core/dtypes/common.py

```diff
@@ -1976,21 +1986,38 @@ def pandas_dtype(dtype):
     Returns
     -------
     np.dtype or a pandas dtype
+
+    Raises
+    ------
+    TypeError if not a dtype
+
     """
 
+    # short-circuit
+    if isinstance(dtype, np.ndarray):
+        return dtype.dtype
```

Comment:
Is this needed? That seems like code smell that you are not sure it is a dtype or an array.

(it should be documented and tested otherwise, this is a public function)